src/Kongverge/DTOs/KongPlugin.cs

```diff
@@ -73,5 +75,45 @@ public virtual Task Validate(IReadOnlyCollection<string> availablePlugins, IColl
         public override bool Equals(object obj) => this.KongEqualsObject(obj);
 
         public override int GetHashCode() => this.GetKongHashCode();
+
+        public override async Task FillInDefaults(IDictionary<string, AsyncLazy<KongPluginSchema>> availablePlugins)
+        {
+            var schema = await availablePlugins[Name].Task;
+            if (Config == null)
+            {
+                Config = new Dictionary<string, object>();
+            }
+            FillInField(schema, Config);
+        }
+
+        private static void FillInField(KongPluginSchema schema, IDictionary<string, object> config)
+        {
+            foreach (var fieldDefinition in schema.Fields)
+            {
+                if (config.ContainsKey(fieldDefinition.Key))
```

Comment:
This section is a took a little parsing.

Would be nice to either wrap the bodies of the clause with descriptive method names (e.g. `CreateDefaultObject` and `FillInBlankDefaults` as examples) or just to have similar comments.)

Reply:
Wilk also add some unit test coverage for this now. It already has coverage by integration tests but that's not explicitly checking this behaviour.